src/test/java/org/cactoos/http/HtWireTest.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.http;
+
+import java.io.IOException;
+import java.net.Socket;
+import java.net.URI;
+import java.util.function.BiFunction;
+import org.cactoos.io.DeadInput;
+import org.cactoos.io.DeadInputStream;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Test case for {@link HtWire}.
+ * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class HtWireTest {
+
+    /**
+     * Default port for HTTP.
+     */
+    private static final int HTTP_PORT = 80;
+
+    /**
+     * Default port for HTTPS.
+     */
+    private static final int HTTPS_PORT = 443;
+
+    @Test
+    public void guessesCorrectPortForHttp() throws IOException {
+        this.checkPorts("http://localhost", HtWireTest.HTTP_PORT);
```

Comment:
@krzyk Please see how other tests work with `hamcrest` in this project. And do the same.

Reply:
@SharplEr Well, this one is not possible with hamcrest as I'm validating that the class called some other object in a correct way by using a mock.